File: CarCareHub_/UI_/flutter_mobile/lib/models/korpa.dart
import 'package:flutter_mobile/models/autoservis.dart';
import 'package:flutter_mobile/models/klijent.dart';
import 'package:flutter_mobile/models/product.dart';
import 'package:flutter_mobile/models/zaposlenik.dart';
import 'package:json_annotation/json_annotation.dart';

part 'korpa.g.dart';

@JsonSerializable()
class Korpa {
  int? korpaId;
  int? proizvodId;
  Product? proizvod;
  int? klijentId;
  Klijent? klijent;
  int? autoservisId;
  Autoservis? autoservis;
  int? zaposlenikId;
  Zaposlenik? zaposlenik;
  int? kolicina;
  double? ukupnaCijenaProizvoda;
  bool? vidljivo;

  Korpa(
      {this.korpaId,
      this.proizvodId,
      this.klijentId,
      this.autoservisId,
      this.zaposlenikId,
      this.kolicina,
      this.ukupnaCijenaProizvoda,
      this.autoservis,
      this.klijent,
      this.proizvod,
      this.zaposlenik,
      this.vidljivo});

  // Factory metoda za deserializaciju JSON-a u Korpa objekt
  factory Korpa.fromJson(Map<String, dynamic> json) => _$KorpaFromJson(json);

  // Metoda za serializaciju Korpa objekta u JSON
  Map<String, dynamic> toJson() => _$KorpaToJson(this);
}

File: CarCareHub_/CarCareHub.Model/Korpa.cs
using System;
using System.Collections.Generic;

namespace CarCareHub.Model
{
    public class Korpa
    {
        public int? ProizvodId { get; set; }

        public int? Kolicina { get; set; }

        public decimal? UkupnaCijenaProizvoda { get; set; }

        public virtual Proizvod? Proizvod { get; set; }

        public int? KlijentId { get; set; }

        public virtual Klijent? Klijent { get; set; }
        public bool Vidljivo { get; set; }
        public int? AutoservisId { get; set; }
        public virtual Autoservis? Autoservis { get; set; }

        public int? ZaposlenikId { get; set; }
        public virtual Zaposlenik? Zaposlenik { get; set; }


        // Metoda za prikazivanje samo ne-null ID-ova
        public Dictionary<string, int?> GetNonNullIds()
        {
            var result = new Dictionary<string, int?>();

            if (KlijentId.HasValue)
                result.Add("KlijentId", KlijentId);

            if (AutoservisId.HasValue)
                result.Add("AutoservisId", AutoservisId);

            if (ZaposlenikId.HasValue)
                result.Add("ZaposlenikId", ZaposlenikId);

            return result;
        }
    }
}
